docs/utilities/feature_flags.md

```diff
@@ -1,57 +1,366 @@
 ---
-title: Feature flags
-description: Utility
+title: Feature flags description: Utility
 ---
 
-The feature flags utility provides a simple rule engine to define when one or multiple features should be enabled depending on the input.
+The feature flags utility provides a simple rule engine to define when one or multiple features should be enabled
+depending on the input.
 
-!!! tip "For simpler use cases where a feature is simply on or off for all users, use [Parameters](parameters.md) utility instead."
+!!! tip "For simpler use cases where a feature is simply on or off for all users, use [Parameters](parameters.md)
+utility instead."
 
 ## Terminology
 
-Feature flags are used to modify a system behaviour without having to change their code. These flags can be static or dynamic.
+Feature flags are used to modify a system behaviour without having to change their code. These flags can be static or
+dynamic.
 
-**Static feature flags** are commonly used for long-lived behaviours that will rarely change, for example `TRACER_ENABLED=True`. These are better suited for [Parameters utility](parameters.md).
+**Static feature flags** are commonly used for long-lived behaviours that will rarely change, for
+example `TRACER_ENABLED=True`. These are better suited for [Parameters utility](parameters.md).
 
-**Dynamic feature flags** are typically used for experiments where you'd want to enable a feature for a limited set of customers, for example A/B testing and Canary releases. These are better suited for this utility, as you can create multiple conditions on whether a feature flag should be `True` or `False`.
+**Dynamic feature flags** are typically used for experiments where you'd want to enable a feature for a limited set of
+customers, for example A/B testing and Canary releases. These are better suited for this utility, as you can create
+multiple conditions on whether a feature flag should be `True` or `False`.
 
-That being said, be mindful that feature flags can increase your application complexity over time if you're not careful; use them sparingly.
+That being said, be mindful that feature flags can increase your application complexity over time if you're not careful;
+use them sparingly.
 
-!!! tip "Read [this article](https://martinfowler.com/articles/feature-toggles.html){target="_blank"} for more details on different types of feature flags and trade-offs"
+!!! tip "Read [this article](https://martinfowler.com/articles/feature-toggles.html){target="_blank"} for more details
```

Comment:
would appreciate a link to my original feature flags blog, but no pressure ;)
https://isenberg-ran.medium.com/aws-lambda-feature-toggles-made-simple-580b0c444233

Reply:
Also a pretty good article https://www.atlassian.com/continuous-delivery/principles/feature-flags

Although Martin Fowler's blog is very comprehensive.

Reply:
CloudBees have their own implementaiton: https://www.cloudbees.com/blog/ultimate-feature-flag-guide

Reply:
ha! I have stumbled upon all of the mentioned articles by now 😅, we can make a list